tests/har.spec.ts

```diff
@@ -311,3 +308,38 @@ it('should have security details', async ({ contextFactory, httpsServer, browser
   else
     expect(securityDetails).toEqual({issuer: 'puppeteer-tests', protocol: 'TLS 1.3', subjectName: 'puppeteer-tests', validFrom: 1550084863, validTo: 33086084863});
 });
+
+it('should have connection details for redirects', async ({ contextFactory, server, browserName }, testInfo) => {
+  server.setRedirect('/foo.html', '/empty.html');
+  const { page, getLog } = await pageWithHar(contextFactory, testInfo);
+  await page.goto(server.PREFIX + '/foo.html');
+  const log = await getLog();
+  expect(log.entries.length).toBe(2);
+
+  const detailsFoo = log.entries[0];
+
+  if (browserName === 'webkit') {
```

Comment:
This appears to be a quirk of WK dev protocol wrt to redirects, atm. If you run the following with `DEBUG="pw:protocol"` you'll only see one `remoteAddress` in the all the logged events (corresponding to the destination, but not the intermediate redirects).

This matches what Safari shows in the Inspector tab.

```js
// @ts-check
const http = require("http");
const pw = require("playwright");

const redirect = http.createServer((req, res) => {
  res.writeHead(301, "Redirect", { Location: "http://localhost:4848"});
  res.end();
});

const server = http.createServer((req, res) => {
  res.setHeader("Content-Type", "text/html");
  res.end("<p>OK</p>");
});

redirect.listen(4747);
server.listen(4848);

(async () => {
  const browser = await pw.webkit.launch();
  const context = await browser.newContext({ recordHar: { path: "record.har" }});
  const page = await browser.newPage();
  await page.goto("http://localhost:4747/foo");
  await page.waitForSelector("text='OK'");
  await context.close();
  await browser.close();
  redirect.close();
  server.close();
})();
```